### PR TITLE
 Add NightLight view

### PR DESCRIPTION
**Pull request by danrabbit**
Todo:
- [x] Only show display tags when the displays tab is open
- [x] Set up URL so we can get directly here from the future indicator

Looks like this:

screenshot from 2017-12-12 10 28 31

### CODE REVIEW

**Comment by cassidyjames** [comment]
@danrabbit does this also only conditionally show up when the night light feature exists, i.e. in artful+?

File: src/DisplayPlug.vala [hunk 1]
```diff
@@ -37,23 +40,72 @@ public class Display.Plug : Switchboard.Plug {
...
-        if (displays_view == null) {
+        if (grid == null) {
             displays_view = new DisplaysView ();
+
+            grid = new Gtk.Grid ();
+            grid.orientation = Gtk.Orientation.VERTICAL;
+
+            var interface_settings_schema = SettingsSchemaSource.get_default ().lookup ("org.gnome.settings-daemon.plugins.color", false);
+            if (interface_settings_schema != null) {
+                if (interface_settings_schema.has_key ("night-light-enabled")) {
```
Comment (tintou) [inline src/DisplayPlug.vala]: maybe in a single line

File: src/DisplayPlug.vala [hunk 2]
```diff
@@ -37,23 +40,72 @@ public class Display.Plug : Switchboard.Plug {
...
         }
 
-        return displays_view;
+        return grid;
     }
 
     public override void shown () {
-        displays_view.displays_overlay.show_windows ();
+        if (stack != null) {
+            if (stack.visible_child == displays_view) {
```
Comment (tintou) [inline src/DisplayPlug.vala]: here it can be in a single line too

File: src/DisplayPlug.vala [hunk 3]
```diff
@@ -37,23 +40,72 @@ public class Display.Plug : Switchboard.Plug {
...
 
     public override void search_callback (string location) {
-        
+        if (stack != null) {
+            if (location == "night-light") {
+                stack.visible_child_name = "night-light";
+            } else {
+                stack.visible_child_name = "displays";
+            }
+            stack.show_all ();
```
Comment (tintou) [inline src/DisplayPlug.vala]: newline after closing bracket

File: src/Views/NightLightView.vala [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        var status_label = new Gtk.Label (_("Night Light:"));
+        status_label.halign = Gtk.Align.END;
+        status_label.xalign = 1;
+
+        var status_switch = new Gtk.Switch ();
+        status_switch.halign = Gtk.Align.START;
+        status_switch.hexpand = true;
+
+        var description_label = new Gtk.Label (_(
+            "Night Light makes the colors of your display warmer. This may help prevent eye strain and sleeplessness."
```
Comment (tintou) [inline src/Views/NightLightView.vala]: I feel like putting the underscore function in the same line than the string would be more elegant

File: src/Views/NightLightView.vala [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        temp_label.margin_top = 24;
+        temp_label.xalign = 1;
+
+        temp_scale = new Gtk.Scale.with_range (Gtk.Orientation.HORIZONTAL, 3500, 6000, 10);
+        temp_scale.draw_value = false;
+        temp_scale.has_origin = false;
+        temp_scale.inverted = true;
+        temp_scale.margin_top = 24;
+        temp_scale.add_mark (3500, Gtk.PositionType.BOTTOM, "More Warm");
+        temp_scale.add_mark (6000, Gtk.PositionType.BOTTOM, "Less Warm");
```
Comment (tintou) [inline src/Views/NightLightView.vala]: Missing localization there

File: src/Views/NightLightView.vala [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    private static DateTime double_date_time (double dbl) {
+        var hours = (int) dbl;
+        var minutes = (int) (dbl - hours) * 100;
+
+        var date_time = new DateTime.local (1, 1, 1, hours, minutes, 0.0);
+
+        if (date_time == null) {
+            warning ("wtf why");
```
Comment (tintou) [inline src/Views/NightLightView.vala]: nope, not in master

**Pull request merged**